### PR TITLE
Option for old unnecessary pings deletion, separate table for player count records

### PR DESCRIPTION
**Pull request by mjezek09**
This PR adds config option for automatic deletion of unnecessary pings, which are older than graph duration. It also moves player count records to separate table to prevent losing those records by pings deletion and to allow easier manipulation.

### CODE REVIEW

File: lib/database.js [hunk 1]
```diff
@@ -200,6 +255,50 @@ class Database {
...
+    })
+    statement.finalize()
+  }
+
+  initOldPingsDelete (callback) {
+    // Delete old ping records on startup
+    logger.info('Deleting old ping records..')
+    this.deleteOldPingRecords(() => {
+      // Delete old ping records every hour
+      setInterval(() => this.deleteOldPingRecords(), 3600000)
```
Comment (hugmanrique) [inline lib/database.js:275]: This interval could be configurable; we could replace the `deleteOldPings` boolean option by an integer `pingsCleanupInterval` that is set to `-1` (disabled) by default.

File: lib/database.js [hunk 2]
```diff
@@ -200,6 +255,50 @@ class Database {
...
+      callback()
+    })
+  }
+
+  deleteOldPingRecords (callback) {
+    // The oldest timestamp that will be kept
+    const oldestTimestamp = TimeTracker.getEpochMillis() - config.graphDuration
+
+    const deleteStart = TimeTracker.getEpochMillis()
+    const statement = this._sql.prepare('DELETE FROM pings WHERE timestamp < ?;')
```
Comment (hugmanrique) [inline lib/database.js:286]: The use of the word "records" here is a bit confusing (due to the newer `player_records` table). Can we rename it to `deleteOldPings` and modify the log messages a bit?

File: lib/database.js [hunk 3]
```diff
@@ -52,6 +52,7 @@ class Database {
 
     this._sql.serialize(() => {
       this._sql.run('CREATE TABLE IF NOT EXISTS pings (timestamp BIGINT NOT NULL, ip TINYTEXT, playerCount MEDIUMINT)', handleError)
+      this._sql.run('CREATE TABLE IF NOT EXISTS players_record (timestamp BIGINT, ip TINYTEXT, playerCount MEDIUMINT)', handleError)
```
Comment (hugmanrique) [inline lib/database.js:55]: If each IP address is meant to have a single record in this table, then I would add a primary key to the `ip` column. That way, we ensure we only keep one record and get nicer performance as a bonus when querying by `ip` (as in the `getRecord` function below).

**Review by hugmanrique — CHANGES_REQUESTED** [review]
Great idea! This will solve the ever-growing database problem. The code looks pretty clean, just left some minor suggestions.

File: lib/database.js (hunk 1 above)
Comment (mjezek09) [reply lib/database.js]: I made separate option for configuring interval, which can also be used to disable interval by setting it to `-1`.

**Review by hugmanrique — APPROVED** [review]

**Comment by Cryptkeeper** [comment]
Thanks for the improvements!

**Pull request merged**